src/main/java/echoquery/sql/Querier.java

```diff
@@ -21,11 +24,98 @@ public Querier(Connection conn) {
+      return Optional.of(new QueryResult(Status.FAILURE,
+          "I'm not sure what table you're interested in, or maybe you referred "
+          + "to one that doesn't exist? Please try again."));
+    }
+
+    if (!request.getAggregationFunc().isPresent()) {
+      return Optional.of(new QueryResult(Status.FAILURE,
+          "I can't tell if you want to know the number of rows or the average, "
+          + "sum, min, or max of a particular column in the "
+          + request.getFromTable().get() + " table. Please try again."));
+    }
+
+    if (!request.getAggregationFunc().get().equals("COUNT")
+        && !request.getAggregationColumn().isPresent()) {
+      return Optional.of(new QueryResult(Status.FAILURE,
+          "I'm not sure what column you want me to find the "
+          + SlotUtil.aggregationFunctionToEnglish(
+              request.getAggregationFunc().get())
+          + " over from the " + request.getFromTable().get()
+          + " table. Please try again."));
+    }
+
+    for (int i = 0; i < request.getComparisonColumns().size(); i++) {
+      String index = "";
+      switch (i) {
+        case 0:
+          index = "first";
+          break;
+        case 1:
+          index = "second";
+          break;
+        case 2:
+          index = "third";
+          break;
+      }
+
+      if (!request.getComparisonColumns().get(i).isPresent()) {
+        return Optional.of(new QueryResult(Status.FAILURE,
+            "I can't tell what column you're interested in filtering on in "
+            + "your " + index + " where clause. Please try again."));
+      }
+      if (!request.getComparators().get(i).isPresent()) {
+        return Optional.of(new QueryResult(Status.FAILURE,
+            "I can't tell if you are looking for values equal to, greater than,"
+            + " or less than the specified value in the "
+            + request.getComparisonColumns().get(i).get() + " column in your "
+            + index + " where clause. Please try again."));
+      }
+      if (!request.getComparisonValues().get(i).isPresent()) {
+        return Optional.of(new QueryResult(Status.FAILURE,
+            "I can't tell if in your " + index + " where clause, you are "
+            + "looking for values in the "
+            + request.getComparisonColumns().get(i).get() + " column"
+            + SlotUtil.comparisonTypeToEnglish(
+                request.getComparators().get(i).get())
+            + "what value. Please try again."));
+      }
+      if (i < request.getComparisonColumns().size() - 1
+          && (i >= request.getComparisonBinaryOperators().size()
+              || !request.getComparisonBinaryOperators().get(i).isPresent())) {
+        return Optional.of(new QueryResult(Status.FAILURE,
+            "It seems like you have more than one where condition but no and "
+            + "or or connecting them, and I'm not sure what to do with this."
+            + " Please try again."));
+      }
+    }
```

Comment:
i'll helper method this, although helper methods without reuse purpose is kind of meh